tests/test_aspa_vector.c

```diff
@@ -0,0 +1,169 @@
+/*
+* This file is part of RTRlib.
+*
+* This file is subject to the terms and conditions of the MIT license.
+* See the file LICENSE in the top level directory for more details.
+*
+* Website; http://rtrlib.realmv6.org/
+*/
+
+#include "rtrlib/lib/vector.h"
+
+#include <assert.h>
+
+static void test_create_vector()
+{
+	struct aspa_vector *vector;
+	assert(aspa_vector_create(&vector) == 0);
+	assert(vector->data != NULL);
+	assert(vector->size == 0);
+	assert(vector->capacity == 128);
+};
+
+static void generate_fake_aspas_record(uint32_t cas, uint32_t random_number, size_t amount,
+				       struct aspas_record **record)
+{
+	struct aspas_record *new_record = (struct aspas_record *)malloc(sizeof(struct aspas_record));
+	new_record->customer_as_number = cas;
+	uint32_t *provider_as_numbers = (uint32_t *)malloc(sizeof(uint32_t) * amount);
```

Comment:
personally, I'd opt for `sizeof(new_record→provider_count[0]) * amount` instead of `sizeof(uint32_t) * amount`

Reply:
I personally find `sizeof(uint32_t) * amount` way more readable.